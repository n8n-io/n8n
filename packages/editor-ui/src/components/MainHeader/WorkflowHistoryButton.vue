<script lang="ts" setup>
import { computed } from 'vue';
import { VIEWS } from '@/constants';
import { useI18n } from '@/composables/useI18n';

const locale = useI18n();

const props = defineProps<{
	workflowId: string;
	isNewWorkflow: boolean;
	isFeatureEnabled: boolean;
}>();

const emit = defineEmits<{
	upgrade: [];
}>();

const workflowHistoryRoute = computed<{ name: string; params: { workflowId: string } }>(() => ({
	name: VIEWS.WORKFLOW_HISTORY,
	params: {
		workflowId: props.workflowId,
	},
}));
</script>

<template>
	<N8nTooltip placement="bottom">
		<RouterLink :to="workflowHistoryRoute" :class="$style.workflowHistoryButton">
			<N8nIconButton
				:disabled="isNewWorkflow || !isFeatureEnabled"
				data-test-id="workflow-history-button"
				type="tertiary"
				icon="history"
				size="medium"
				text
			/>
		</RouterLink>
		<template #content>
			<span v-if="isFeatureEnabled && isNewWorkflow">
				{{ locale.baseText('workflowHistory.button.tooltip.empty') }}
			</span>
			<span v-else-if="isFeatureEnabled">{{
				locale.baseText('workflowHistory.button.tooltip.enabled')
			}}</span>
			<i18n-t v-else keypath="workflowHistory.button.tooltip.disabled">
				<template #link>
					<N8nLink size="small" @click="emit('upgrade')">
						{{ locale.baseText('workflowHistory.button.tooltip.disabled.link') }}
					</N8nLink>
				</template>
			</i18n-t>
		</template>
	</N8nTooltip>
</template>

<style lang="scss" module>
.workflowHistoryButton {
	width: 30px;
	height: 30px;
	color: var(--color-text-dark);
	border-radius: var(--border-radius-base);

	&:hover {
		background-color: var(--color-background-base);
	}

	:disabled {
		background: transparent;
		border: none;
		opacity: 0.5;
	}
}
</style>
