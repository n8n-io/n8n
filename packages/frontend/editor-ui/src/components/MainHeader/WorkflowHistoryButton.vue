<script lang="ts" setup>
import { computed } from 'vue';
import { VIEWS } from '@/constants';
import { useI18n } from '@n8n/i18n';
import { I18nT } from 'vue-i18n';

const locale = useI18n();

const props = defineProps<{
	workflowId: string;
	isNewWorkflow: boolean;
	isFeatureEnabled: boolean;
}>();

const emit = defineEmits<{
	upgrade: [];
}>();

const workflowHistoryRoute = computed<{ name: string; params: { workflowId: string } }>(() => ({
	name: VIEWS.WORKFLOW_HISTORY,
	params: {
		workflowId: props.workflowId,
	},
}));
</script>

<template>
	<N8nTooltip placement="bottom">
		<RouterLink :to="workflowHistoryRoute" :class="$style.workflowHistoryButton">
			<N8nIconButton
				:disabled="isNewWorkflow || !isFeatureEnabled"
				data-test-id="workflow-history-button"
				type="tertiary"
				icon="history"
				size="medium"
				text
			/>
		</RouterLink>
		<template #content>
			<span v-if="isFeatureEnabled && isNewWorkflow">
				{{ locale.baseText('workflowHistory.button.tooltip.empty') }}
			</span>
			<span v-else-if="isFeatureEnabled">{{
				locale.baseText('workflowHistory.button.tooltip.enabled')
			}}</span>
			<I18nT v-else keypath="workflowHistory.button.tooltip.disabled" scope="global">
				<template #link>
					<N8nLink size="small" @click="emit('upgrade')">
						{{ locale.baseText('workflowHistory.button.tooltip.disabled.link') }}
					</N8nLink>
				</template>
			</I18nT>
		</template>
	</N8nTooltip>
</template>

<style lang="scss" module>
.workflowHistoryButton {
	width: 30px;
	height: 30px;
	color: var(--color-text-dark);
	border-radius: var(--border-radius-base);

	&:hover {
		background-color: var(--color-background-base);
	}

	:disabled {
		background: transparent;
		border: none;
		opacity: 0.5;
	}
}
</style>
