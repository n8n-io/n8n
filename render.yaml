services:
  - type: web
    name: luci-n8n-agent
    env: docker
    plan: free
    branch: master
    autoDeploy: true
    buildCommand: ""
    startCommand: "n8n start"
    envVars:
      - key: N8N_BASIC_AUTH_ACTIVE
        value: "true"
      - key: N8N_BASIC_AUTH_USER
        value: "admin"
      - key: N8N_BASIC_AUTH_PASSWORD
        value: "admin"
      - key: N8N_HOST
        value: "0.0.0.0"
      - key: N8N_PORT
        value: "5678"
      - key: WEBHOOK_URL
        value: https://luci-n8n-agent.onrender.com/
