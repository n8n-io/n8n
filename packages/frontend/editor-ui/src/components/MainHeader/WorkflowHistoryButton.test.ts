import { createComponentRenderer } from '@/__tests__/render';
import { within } from '@testing-library/vue';
import userEvent from '@testing-library/user-event';
import WorkflowHistoryButton from '@/components/MainHeader/WorkflowHistoryButton.vue';

vi.mock('vue-router', () => ({
	useRoute: () => vi.fn(),
	useRouter: () => vi.fn(),
	RouterLink: vi.fn(),
}));

const renderComponent = createComponentRenderer(WorkflowHistoryButton, {
	global: {
		stubs: {
			RouterLink: true,
			'router-link': {
				template: '<div><slot /></div>',
			},
			N8nTooltip: {
				template: '<div><slot /><slot name="content" /></div>',
			},
			N8nIconButton: true,
			N8nLink: {
				template: '<a @click="$emit(\'click\')"><slot /></a>',
			},
			I18nT: {
				template: '<span><slot name="link" /></span>',
			},
		},
	},
});

describe('WorkflowHistoryButton', () => {
	it('should be disabled if the feature is disabled', async () => {
		const { getByRole, queryByTestId, emitted } = renderComponent({
			props: {
				workflowId: '1',
				isNewWorkflow: false,
				isFeatureEnabled: false,
			},
		});
		const button = queryByTestId('workflow-history-button');
		expect(button).toHaveAttribute('disabled', 'true');
		if (!button) {
			throw new Error('Button does not exist');
		}
		await userEvent.hover(button);
		expect(getByRole('tooltip')).toBeVisible();

		within(getByRole('tooltip')).getByText('View plans').click();

		expect(emitted()).toHaveProperty('upgrade');
	});

	it('should be disabled if the feature is enabled but the workflow is new', async () => {
		const { queryByTestId } = renderComponent({
			props: {
				workflowId: '1',
				isNewWorkflow: true,
				isFeatureEnabled: true,
			},
		});
		expect(queryByTestId('workflow-history-button')).toHaveAttribute('disabled', 'true');
	});

	it('should be enabled if the feature is enabled and the workflow is not new', async () => {
		const { container, queryByTestId } = renderComponent({
			props: {
				workflowId: '1',
				isNewWorkflow: false,
				isFeatureEnabled: true,
			},
		});
		expect(queryByTestId('workflow-history-button')).toHaveAttribute('disabled', 'false');
		expect(container).toMatchSnapshot();
	});
});
