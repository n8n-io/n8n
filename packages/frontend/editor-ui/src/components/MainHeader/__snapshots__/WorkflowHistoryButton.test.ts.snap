// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WorkflowHistoryButton > should be enabled if the feature is enabled and the workflow is not new 1`] = `
<div>
  
  <div
    class="el-tooltip__trigger"
    to="[object Object]"
  >
    
    <n8n-icon-button-stub
      active="false"
      data-test-id="workflow-history-button"
      disabled="false"
      icon="history"
      loading="false"
      outline="false"
      size="medium"
      text="false"
      type="highlight"
    />
    
  </div>
  <!--teleport start-->
  <!--teleport end-->
  
</div>
`;
